scripts/justmat/foulplay.lua

```diff
@@ -78,14 +87,29 @@ end
 
 local function trig()
   for i=1,8 do
```

Comment:
an alternative would be to iterate over the track table something like this:

```lua
for i, t in ipairs(track) do
  if t.trig_logic==0 and t.s[t.pos] then
    if math.random(100) <= t.prob then
      engine.trig(i-1)
    end
    ...
end
```

Reply:
I had not seen ipairs before, and had to look it up :)  I especially like how much more readable this seems. These are likely changes I will make before my next PR.

Reply:
good deal!  another related tip is that in case you don't need the index, it's convention to name it `_` (that's generally true for ignored parameters).   something like:

```lua
for _, t in ipairs(track) do ...
```

in this case you do of course but maybe not in other loops?

oh and there's also the `pairs` function.  here's a bit about [the difference](http://www.luafaq.org/#T1.10).

Reply:
@pq - I rewrote the track table setup, and the trig function today, and these tips really made things nicer. With the added readability, implementing the other logical trigger conditions was much easier! I will test for a day or two then update the docs and PR.

thanks again!

Reply:
awesome!  you're more than welcome.  look forwarding to seeing where this takes you!